spinalcordtoolbox/gui/centerline.py

```diff
@@ -23,39 +23,35 @@ class CenterlineController(base.BaseController):
     _mode = ''
     INTERVAL = 15
     MODES = ['AUTO', 'CUSTOM']
-    _slice = 0.0
 
     def __init__(self, image, params, init_values=None):
         super(CenterlineController, self).__init__(image, params, init_values)
 
     def reformat_image(self):
-        max_z = self.image.dim[2]
         super(CenterlineController, self).reformat_image()
+        max_x, _, max_z = self.image.dim[:3]
         self.params.num_points = self.params.num_points or 11
 
-        if self.params.starting_slice < 0:
+        # if the starting slice is of invalid value then use default value
+        if self.params.starting_slice > max_x or self.params.starting_slice < 0:
             self.params.starting_slice = self.default_position[0]
         # if the starting slice is a fraction, recalculate the starting slice as a ratio.
         elif 0 < self.params.starting_slice < 1:
             self.params.starting_slice = max_z // self.params.starting_slice
 
-        self._slice = self.params.starting_slice
-        self.INTERVAL = (max_z - self.params.starting_slice) // (self.params.num_points - 1) or 3
+        self.INTERVAL = (max_x - self.params.starting_slice) // (self.params.num_points - 1) or 3
         self.reset_position()
 
     def reset_position(self):
         super(CenterlineController, self).reset_position()
         if self.mode == 'AUTO':
-            self.position = (self._slice, self.position[1], self.position[2])
+            self.position = (0, self.position[1], self.position[2])
```

Comment:
Maybe a [`namedtuple`](https://docs.python.org/2/library/collections.html#collections.namedtuple) would be better here, like in:

`Position = namedtuple('Position', ['x', 'y', 'z'])`

Then the position name will be implicit, such as:

`p = Position(x=1, y=2, z=3)`

Which is more meaninful than:

`p = (1, 2, 3)`

Where we don't know the semantics of every tuple position.